### PR TITLE
[ChangeFeedProcessor]ExhaustAllRetriesBeforePollDelay

### PR DESCRIPTION
**Pull request by xinlian12**
Fixes issue https://github.com/Azure/azure-sdk-for-java/issues/34901

**Issue:**
Expected behavior:

- Read the change feed.
- If there are no changes, sleep for a predefined amount of time (customizable with options.setFeedPollDelay in the Builder) and go to step 1
- If there are changes, send them to the delegate.
- When the delegate finishes processing the changes successfully, update the lease store with the latest processed point in time and go to step 1

Current behavior:
- Read the change feed.
- If there are no changes, sleep for a predefined amount of time (customizable with options.setFeedPollDelay in the Builder) and go to step 1
- If there are changes, send them to the delegate.
- When the delegate finishes processing the changes successfully, update the lease store with the latest processed point in time and go to step 1
- Sleep for predefined amount of time then go to step 1.

**Fixes:**
Keep fetching left pages until response is 304(Not_Modified), then sleep for pollDelay time

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@
 #### Breaking Changes
 
 #### Bugs Fixed
+* Fixed an issue for `ChangeFeedProcessor` that it should fetch all changes before delay the fetch based on configured `PollDelay`. - See [PR 35324](https://github.com/Azure/azure-sdk-for-java/pull/35324)
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/CHANGELOG.md:13]:
```suggestion
* Fixed an issue with `ChangeFeedProcessor` to fetch all changes before delay based on configured `PollDelay`. - See [PR 35324](https://github.com/Azure/azure-sdk-for-java/pull/35324)
```

**Review by kushagraThapar — APPROVED** [review]
LGTM, thanks @xinlian12 - one minor suggestion.

File: sdk/cosmos/azure-cosmos/CHANGELOG.md (hunk 1 above)
Comment (xinlian12) [reply sdk/cosmos/azure-cosmos/CHANGELOG.md]: updated

**Review by milismsft — COMMENTED** [review]
looks good; thank you for adding the extensive E2E tests

**Pull request merged**